### PR TITLE
virsh_hostname: Add test to connect to a remote hypervisor

### PR DESCRIPTION
**Pull request by hahakiki2010**
Signed-off-by: Yan Li <yannli@redhat.com>

### CODE REVIEW

**Comment by hahakiki2010** [comment]
Test passed
(virtenv_py3) [root@localhost host]# avocado run --vt-type libvirt virsh.hostname
JOB ID     : 94bc6789be18af49557c8010ceeadd0312fc4fab
JOB LOG    : /root/avocado/job-results/job-2018-07-12T14.38-94bc678/job.log
 (1/4) type_specific.io-github-autotest-libvirt.virsh.hostname.no_option: PASS (3.39 s)
 (2/4) type_specific.io-github-autotest-libvirt.virsh.hostname.unexpect_option: PASS (6.96 s)
 (3/4) type_specific.io-github-autotest-libvirt.virsh.hostname.with_libvirtd_stop: PASS (4.00 s)
 (4/4) type_specific.io-github-autotest-libvirt.virsh.hostname.remote_connect: PASS (23.24 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 69.58 s

File: libvirt/tests/src/virsh_cmd/host/virsh_hostname.py [hunk 1]
```diff
@@ -13,19 +16,41 @@ def run(test, params, env):
...
+        if not utils_package.package_install("libvirt", session):
+            test.cancel("Failed to install libvirt on remote server")
+        libvirtd = utils_libvirtd.Libvirtd(session=session)
+        libvirtd.restart()
+
     # Run test case
     option = params.get("virsh_hostname_options")
-    hostname_test = virsh.hostname(option,
+    remote_ip = params.get("remote_ip", "None")
+    if remote_ip.count("EXAMPLE"):
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_hostname.py]:
```python
remote_ip = params.get("remote_ip", None)
if not remote_ip or remote_ip.count("EXAMPLE"):
```

Comment (hahakiki2010) [reply libvirt/tests/src/virsh_cmd/host/virsh_hostname.py]: fixed

**Comment by hahakiki2010** [comment]
@balamuruhans Comments fixed. Thanks

File: libvirt/tests/src/virsh_cmd/host/virsh_hostname.py [hunk 2]
```diff
@@ -13,19 +16,42 @@ def run(test, params, env):
...
+    if remote_uri:
+        if not utils_package.package_install("libvirt", session):
+            test.cancel("Failed to install libvirt on remote server")
+        libvirtd = utils_libvirtd.Libvirtd(session=session)
+        libvirtd.restart()
+
     # Run test case
     option = params.get("virsh_hostname_options")
-    hostname_test = virsh.hostname(option,
+    remote_ip = params.get("remote_ip", "None")
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_hostname.py]: please make the default value to `None`, not use as str

Comment (hahakiki2010) [reply libvirt/tests/src/virsh_cmd/host/virsh_hostname.py]: bad typo, fixed. Thanks @balamuruhans

**Comment by balamuruhans** [comment]
Thanks @hahakiki2010, let's merge it.

**Pull request merged**